tokio-threadpool/src/thread_pool.rs

```diff
@@ -64,6 +83,49 @@ impl ThreadPool {
         self.sender().spawn(future).unwrap();
     }
 
+    /// Spawn a future on to the thread pool, return a future representing 
+    /// the produced value.
+    /// 
+    /// The SpawnHandle returned is a future that is a proxy for future itself. 
+    /// When future completes on this thread pool then the SpawnHandle will itself 
+    /// be resolved.
+    /// 
+    /// # Examples
+    ///
+    /// ```rust
+    /// # extern crate tokio_threadpool;
+    /// # extern crate futures;
+    /// # use tokio_threadpool::ThreadPool;
+    /// use futures::future::{Future, lazy};
+    ///
+    /// # pub fn main() {
+    /// // Create a thread pool with default configuration values
+    /// let thread_pool = ThreadPool::new();
+    ///
+    /// let handle = thread_pool.spawn_handle(lazy(|| -> Result<_, ()> {
+    ///     Ok(42)
```

Comment:
Another way to make this work is to return `Ok::<_, ()>(42)`, but this works too.